### PR TITLE
Add regrets-reporter schemas

### PR DESCRIPTION
**Pull request by motin**
... under the telemetry and regrets-reporter namespaces (one for browser.telemetry.submitPing and one for a custom telemetry client)

### CODE REVIEW

File: schemas/regrets-reporter/regrets-reporter-update/regrets-reporter-update.1.schema.json [hunk 1]
```diff
@@ -0,0 +1,230 @@
+{
+  "$comment": "See https://docs.google.com/document/d/104x-QmODN7CQwvKVDlB4D2BJzJ0Q-BKnluW5EA7vX7Q/edit",
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "definitions": {
+    "AmountsByCategory": {
```
Comment (acmiyaguchi) [inline schemas/regrets-reporter/regrets-reporter-update/regrets-reporter-update.1.schema.json:5]: I regret to inform you that the tooling used to generate BigQuery tables from jsonschema doesn't support definitions. The convention we use for templating in this case is to define the subset of the schemas under `template/include/<namespace>/<definition>.<version>.schema.json`. You would then call it using cmake includes e.g.

```json
...
        "amount_of_days_of_at_least_one_video_played_on_a_youtube_watch_page": {
          "$ref": "#/definitions/AmountsByCategory"
        },
...
```

would turn into a `templates/include/regrets-reporter/amountsByCategory.1.schema.json`

```
      "properties": {
        "in_total": {
          "type": "number"
        },
        "via_non_search_algorithmic_recommendations_content": {
          "type": "number"
        },
        "via_recommendations_with_an_explicit_query_or_constraint_to_optimize_for": {
          "type": "number"
        },
        "via_search_results": {
          "type": "number"
        }
      },
      "type": "object"
```

And would be called via:

```json
...
        "amount_of_days_of_at_least_one_video_played_on_a_youtube_watch_page": {
          @REGRETS_REPORTER_AMOUNTS_BY_CATEGORY_1_JSON@
        },
...
```

Comment (motin) [reply schemas/regrets-reporter/regrets-reporter-update/regrets-reporter-update.1.schema.json:5]: Thanks for the heads up. Fixed now :) Btw I found the working syntax to be `@REGRETS-REPORTER_AMOUNTSBYCATEGORY_1_JSON@`

File: templates/telemetry/regrets-reporter-update/regrets-reporter-update.1.schema.json [hunk 2]
```diff
@@ -0,0 +1 @@
+@REGRETS-REPORTER_REGRETSREPORTERUPDATE_1_JSON@
```
Comment (acmiyaguchi) [inline templates/telemetry/regrets-reporter-update/regrets-reporter-update.1.schema.json:1]: I believe it's still the case that the document version for pings sent through telemetry must be set to 4. This file should be changed to be `telemetry/regrets-reporter-update/regrets-reporter-update.4.schema.json`

**Comment by mreid-moz** [comment]
@motin can you link to the corresponding Data Collection Review bug?

**Comment by motin** [comment]
> @motin can you link to the corresponding Data Collection Review bug?

https://bugzilla.mozilla.org/show_bug.cgi?id=1644107#c2

**Comment by whd** [comment]
Definitely do not merge this. I don't have access to the data review bug (perhaps telling) but given @mreid-moz's recent questions, this data (if we do end up collecting it) may require special provisioning logic that I will need to prepare before this PR is merged.

**Comment by motin** [comment]
I don't have access to the CI logs, so I don't know why the tests are failing. Running `scripts/mps-build`, then `scripts/mps-test` results in passing tests locally:

```
$ scripts/mps-test
================================================ test session starts =================================================
platform linux -- Python 3.6.8, pytest-5.3.5, py-1.8.1, pluggy-0.13.1
rootdir: /app
plugins: xdist-1.31.0, forked-1.1.3
collected 374 items

tests/test_validation_java.py ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 21%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss    [ 50%]
tests/test_validation_python.py .............................
................................................. [ 70%]
.............................................................................................................  [100%]

==================================== 187 passed, 187 skipped in 206.02s (0:03:26) ====================================
```

**Comment by acmiyaguchi** [comment]
The failure was coming from the BigQuery diff script (`docker run -w /app mps pytest scripts/bigquery_schema_diff.py`), but it looks like you might have resolved this now.

**Comment by motin** [comment]
@mreid-moz @acmiyaguchi Data review is now r+ https://bugzilla.mozilla.org/show_bug.cgi?id=1644107#c17

**Comment by whd** [comment]
This is still blocked on some policy decisions mapping categories in https://wiki.mozilla.org/Firefox/Data_Collection#Data_Collection_Categories to actual pipeline configuration, which I've filed https://bugzilla.mozilla.org/show_bug.cgi?id=1651005 to address.

**Comment by mreid-moz** [comment]
This has been approved for collection during the _internal staff testing period only_. As such we are unblocked to merge this PR. I have filed [Bug 1654078](https://bugzilla.mozilla.org/show_bug.cgi?id=1654078) to capture the work that must happen before collecting data any more broadly.

Re-r? @acmiyaguchi to make sure the schema itself looks good.

File: schemas/telemetry/regrets-reporter-update/regrets-reporter-update.4.schema.json [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+        "report_data": {
+          "properties": {
+            "parent_youtube_navigations_metadata": {
+              "items": {
+                "description": "Selected/derived attributes from YouTubeNavigations",
+                "properties": {
+                  "document_visible_time": {
+                    "type": "number"
+                  },
+                  "page_entry_point": {},
```
Comment (acmiyaguchi) [inline schemas/telemetry/regrets-reporter-update/regrets-reporter-update.4.schema.json:67]: The generated schema is missing the page entry point.

Comment (motin) [reply schemas/telemetry/regrets-reporter-update/regrets-reporter-update.4.schema.json]: Thanks @acmiyaguchi I missed this during a recent update, and as @mreid-moz suggested offline, a file was improperly named. This has now been addressed.

**Pull request merged**